File: ChickInRun/Assets/floormap/Gate/lavagate_lava.js
#pragma strict

public var openMat : Material;
public var openNumber : int;
private var lavaNum : int;
private var lGate : GameObject;
//gate의 변수를 받아오기 위한 오브젝트

function Start () {
	lGate = GameObject.FindGameObjectWithTag("LavaGate");
}

function Update () {

	lavaNum = lGate.GetComponent(GateOpen).openNum;
	if(lavaNum == openNumber){
		GetComponent.<Renderer>().material = openMat;
	}
}

File: ChickInRun/Assets/Script/LavaGateLava.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class LavaGateLava : MonoBehaviour
{
    public Material openMat;
    public int openNumber;
    private int lavanum;
    private GameObject lGate;

    // Start is called before the first frame update
    void Start()
    {
        lGate = GameObject.FindGameObjectWithTag("LavaGate");
    }

    // Update is called once per frame
    void Update()
    {
        lavanum = lGate.GetComponent<GateOpen>().openNum;
        if(lavanum == openNumber)
        {
            GetComponent<Renderer>().material = openMat;
        }
    }
}
